### PR TITLE
Fogl 527

### PR DESCRIPTION
**Pull request by vaibhav-ScaleDB**
Implemented RESTAPI using Mocks. Actual implementation after FOGL-531 is in place in a separate pull

### CODE REVIEW

File: src/python/foglamp/core/api/backup_restore.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    :Example: curl -X GET  http://localhost:8082/foglamp/backup
+    :Example: curl -X GET  http://localhost:8082/foglamp/backup?limit=2&skip=1&status=complete
+    """
+    try:
+        limit = request.query['limit'] if 'limit' in request.query else None
+        skip = request.query['skip'] if 'skip' in request.query else None
+        status = request.query['status'] if 'status' in request.query else None
+
+        if status and status not in ['complete', 'running', 'failed']:
+            return web.json_response({'error': 'Incorrect status'})
```
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api/backup_restore.py]: better to raise a web.HTTPBadRequest exception here

File: src/python/foglamp/core/api/backup_restore.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+        if limit:
+            try:
+                limit = int(limit)
+            except ValueError:
+                return web.json_response({'error': 'Limit can be a positive integer only'})
```
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api/backup_restore.py]: better to raise a web.HTTPBadRequest exception here

File: src/python/foglamp/core/api/backup_restore.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+        if skip:
+            try:
+                skip = int(skip)
+            except ValueError:
+                return web.json_response({'error': 'Skip can be a positive integer only'})
```
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api/backup_restore.py]: better to raise a web.HTTPBadRequest exception here

**Comment by amarendra-dianomic** [comment]
@vaibhav-ScaleDB Please wait for merger of PR#187 for FOGL-509, then the above review comments will become more clear.

**Comment by praveen-garg** [comment]
add the tests as well

**Comment by vaibhav-ScaleDB** [comment]
Test skeleton added

**Comment by sstefano69** [comment]
the code is fine on my side, we will have to implement the methods for the integration within the Backup/Restore classes.

**Pull request merged**